## Junit5 테스트 코드

- 에러 메세지의 종류까지 테스트하기

``` 
package baseball.util;


import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

class ValidatorTest {

    @Nested
    class invalidInput {

        @DisplayName("입력 범위 초과")
        @ParameterizedTest
        @ValueSource(strings = {"222222222222222222", "12731267764214627128721763"})
        void 입력_범위_초과(String input) {
            assertThatThrownBy(() -> Validator.getValidatedPlayerNumber(input))
                    .isInstanceOf(IllegalArgumentException.class)
                    .hasMessage(ExceptionMessage.OUT_OF_RANGE.getMessage());
        }
    }

}
```

- 에러를 발생시키지 않음

``` 
@Nested
class validInputTest {
    @ParameterizedTest
    @ValueSource(strings = {"123", "213", "262", "111"})
    void 정상_입력(String input) {
        assertThatCode(() -> Validator.getValidatedPlayerNumber(input))
                .doesNotThrowAnyException();
    }
}
```